src/core/lib/security/security_connector/security_connector.cc

```diff
@@ -465,12 +465,10 @@ static bool fake_channel_check_call_host(grpc_channel_security_connector* sc,
                                          grpc_error** error) {
   grpc_fake_channel_security_connector* c =
       reinterpret_cast<grpc_fake_channel_security_connector*>(sc);
-  if (c->is_lb_channel) {
-    // TODO(dgq): verify that the host (ie, authority header) matches that of
-    // the LB, as opposed to that of the backends.
-  } else {
-    // TODO(dgq): verify that the host (ie, authority header) matches that of
-    // the backend, not the LB's.
+  if (c->is_lb_channel && strcmp(host, c->target) != 0) {
+    gpr_log(GPR_ERROR, "Authority (host) '%s' != Target '%s' for an LB call",
+            host, c->target);
+    abort();
```

Comment:
It's not clear to me what should be checked in the case of a `host` being specified when calling `grpc_channel_create_call`. In this case, `c->target`, set at channel creation, can be different from the `host`, specified at call creation. It was only clear what should happen in the LB channel/call case. @yihuazhang any ideas?

Reply:
It seems in grpclb usecase, those two are always the same, so we are fine. But In generic usecase, I think if a caller specifies a different host info from c->target to grpc_channel_create_call(), we should fail check_call_host(), or find a way to do another secure naming check on the host info. My question is, is it possible (or is there any usecase) that a client creates a channel to a gRPC service (by specifying a target info), and he can also call grpc_channel_create_call() during the operation to modify the host info?

Reply:
Right now, several tests perform a host override when creating calls (see [here](https://github.com/grpc/grpc/blob/59ea0ae3ebcca0aef5a15c5aa5b4d27b7f3fc9c4/test/core/end2end/tests/cancel_after_round_trip.cc#L153), for example). For all tests run under [`h2_fakesec.cc`](https://github.com/grpc/grpc/blob/master/test/core/end2end/fixtures/h2_fakesec.cc), which simply plugs in the fake security, this will make `fake_channel_check_call_host()` fail. Given that the tests use "foo.test.google.fr:1234" as the call host override but the actual target will be something like "localhost:34253", I can only think of passing the call host override expectation to the fake security connector when creating it.

Reply:
It seems besides the tests, there are no other callers of get_host_override_slice(). Can we confirm from this, the host override will not happen in real workloads?